cpp/src/arrow/compute/kernels/aggregate_basic_internal.h

```diff
@@ -322,6 +411,7 @@ struct MinMaxImpl : public ScalarAggregator {
     StateType local;
     local.has_nulls = !scalar.is_valid;
     local.has_values = scalar.is_valid;
+    this->count += scalar.is_valid;
```

Comment:
`has_values` is a bit redundant now that the count of valid values is tracked, isn't it?